microsoft/testsuites/vm_extensions/kvvm_extension.py

```diff
@@ -0,0 +1,201 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import os
+import random
+import time
+from typing import List
+
+from assertpy import assert_that
+
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.operating_system import BSD
+from lisa.sut_orchestrator.azure.common import (
+    add_system_assign_identity,
+    check_certificate_existence,
+    check_system_status,
+    create_certificate,
+    create_keyvault,
+    delete_certificate,
+    delete_keyvault,
+    get_node_context,
+    rotate_certificates,
+)
+from lisa.sut_orchestrator.azure.features import AzureExtension
+from lisa.sut_orchestrator.azure.platform_ import AzurePlatform
+from lisa.testsuite import TestResult
+
+
+@TestSuiteMetadata(
+    area="vm_extension",
+    category="functional",
+    description="BVT for Azure Key Vault Extension",
+    requirement=simple_requirement(unsupported_os=[]),
+)
+class AzureKeyVaultExtensionBvt(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        The following test case validates the Azure Key Vault Linux
+        Extension while creating the following resources:
+        A resource group
+        A VM
+        A Key Vault
+        Two certificates in the Key Vault
+        Retrieval of the certificate's secrets
+        through SecretClient class from Azure SDK.
+        Installation of the Azure Key Vault Linux Extension on the VM.
+        Rotation of the certificates (After KVVM Extension has been installed)
+        All of the resources have been created by using the Azure SDK Python.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            supported_features=[AzureExtension], unsupported_os=[BSD]
+        ),
+    )
+    def verify_key_vault_extension(
+        self, log: Logger, node: Node, result: TestResult
+    ) -> None:
+        # Section for vault name and supported OS check
+        vault_name = f"python-keyvault-{random.randint(1, 100000):05}prp"
```

Comment:
Make the kv name associated with the test case name like `kv-extension-...`

Reply:
The test case may be run more than 10k times in a month, it's easy to be conflict. Please use a shared key vault to avoid conflict.